clients/python-apiserver-client/python_apiserver_client/params/jobsubmission.py

```diff
@@ -0,0 +1,108 @@
+        self.num_gpu = num_gpu
+        self.resources = resources
+
+    def to_string(self) -> str:
+        val = f"entrypoint = {self.entrypoint}"
+        if self.submission_id is not None:
+            val += f", submission_id = {self.submission_id}"
+        if self.num_cpu > 0:
+            val += f", num_cpu = {self.num_cpu}"
+        if self.num_gpu > 0:
+            val += f", num_gpu = {self.num_gpu}"
+        if self.runtime_env is not None:
+            val += f", runtime_env = {self.runtime_env}"
+        if self.metadata is not None:
+            val += f", metadata = {self.metadata}"
+        if self.resources is not None:
+            val += f", resources = {self.resources}"
+        return val
+
+    def to_dict(self) -> dict[str, any]:
+        dct = {"entrypoint": self.entrypoint}
+        if self.submission_id is not None:
+            dct["submissionId"] = self.submission_id
+        if self.runtime_env is not None:
+            dct["runtimeEnv"] = self.runtime_env
+        if self.metadata is not None:
+            dct["metadata"] = self.metadata
+        if self.num_cpu > 0:
+            dct["numCpus"] = self.num_cpu
+        if self.num_gpu > 0:
+            dct["numGpus"] = self.num_gpu
+        if self.resources is not None:
+            dct["resources"] = self.resources
+        return dct
+
+
+class RayJobInfo:
+    """
+    RayJobInfo used to define information about the job in a Ray cluster
+    It provides APIs to create and stringify. Its output only data, so we do not need to implement to_dict
+
+    Methods:
+    - Create RayJobRequest: gets the following parameters:
+        entrypoint - the command to start a job on the cluster
+        job_id - job execution id
+        submission_id - submission id for the job submission
+        runtime_env - job runtime environment
+        status - job execution status
+        message - status message
+        start_time - job start time
+        end-time - job ind time
+        error_type - type of error
+        metadata - optional, dictionary of the submission metadata
+    """
+    def __init__(self, dst: dict[str, any]) -> None:
+        self.entrypoint = dst.get("entrypoint", "")
+        self.job_id = dst.get("jobId", "")
+        self.submission_id = dst.get("submissionId", "")
+        self.status = dst.get("status", "")
+        self.message = dst.get("message", None)
+        self.start_time = int(dst.get("startTime", "0"))
+        self.end_time = int(dst.get("endTime", "0"))
+        self.error_type = dst.get("ErrorType", None)
+        self.metadata = dst.get("Metadata", None)
+        self.runtime_env = dst.get("runtimeEnv", None)
+
+    def to_string(self) -> str:
+        val = (f"entrypoint = {self.entrypoint}, job id {self.job_id}, submission id = {self.submission_id},"
+               f" status = {self.status}")
+        if self.message is not None:
+            val += f" message = {self.message}"
+        if self.start_teme > 0:
```

Comment:
```suggestion
        if self.start_teme > 0:
```
@blublinsky We still need to change this one to `start_time` and all the other occurrences, otherwise I think it will error.  Rhe unit test should have caught this.

Reply:
I see we are calling `RayJobInfo.to_string` in the unit test, so I'm confused how it could be passing.  Is is possible that it's not running in CI?

Reply:
fixed